crates/reth/evm/src/precompiles/schnorr.rs

```diff
@@ -0,0 +1,53 @@
+use revm::{
+    precompile::utilities::right_pad,
+    primitives::{PrecompileOutput, PrecompileResult},
+};
+use revm_primitives::Bytes;
+use strata_crypto::verify_schnorr_sig;
+use strata_primitives::buf::{Buf32, Buf64};
+use tracing::warn;
+
+/// Internal representation of parsed Schnorr input bytes.
+struct SchnorrInput {
+    /// 32 Bytes: Public key
+    public_key: Buf32,
+    /// 32 Bytes: Message hash
+    message_hash: Buf32,
+    /// 64 Bytes: Schnorr Signature
+    signature: Buf64,
+}
+
+fn parse_schnorr_input(input: &Bytes) -> SchnorrInput {
+    if input.len() < 128 {
+        warn!(
+            input_len = %input.len(),
+            "Input length ({}) is below the minimum required threshold of 128 bytes",
+            input.len()
+        );
+    }
```

Comment:
I would say that since we're defining away the error case case we don't need this warning here.  We don't control how this gets called anyways, so we could have people spamming logs with warnings by sending txs that verify malformed signatures and repeatedly RBFing them.  This also saves us from having to import `tracing` (and figure out how to use it in the proofs, I'm not sure how that'd even work).

We should just have unit tests for this to ensure that message validation fails as we'd expect it to (if the signature doesn't end in zeroes I guess).

I realize I actually contradicted myself in those two comments, I read the reth code afterwards.